File: backend/app/Models/BusinessScreening.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class BusinessScreening extends Model
{
    use HasFactory;

    protected $fillable = [
        'company_id',
        'ticker',
        'business_summary',
        'current_core_business',
        'detected_business_activities',
        'detected_prohibited_activities',
        'supporting_evidence',
        'source_urls',
        'source_publication_dates',
        'ai_explanation',
        'confidence_score',
        'business_compliance_status',
        'last_analysed_timestamp',
    ];

    protected $casts = [
        'detected_business_activities' => 'array',
        'detected_prohibited_activities' => 'array',
        'supporting_evidence' => 'array',
        'source_urls' => 'array',
        'source_publication_dates' => 'array',
        'confidence_score' => 'float',
        'last_analysed_timestamp' => 'datetime',
    ];

    public function company()
    {
        return $this->belongsTo(Company::class);
    }
}

File: backend/ai-engine/app/models/business_screening.py
import uuid
from sqlalchemy import Column, String, Integer, Float, DateTime, func, Text, ForeignKey
from sqlalchemy.dialects.postgresql import UUID, JSONB
from app.core.database import Base

class BusinessScreening(Base):
    __tablename__ = "business_screenings"

    id = Column(Integer, primary_key=True, index=True)
    company_id = Column(Integer, index=True) # Usually we'd map this but for simplicity we rely on ticker
    ticker = Column(String, index=True, nullable=False)
    
    business_summary = Column(Text, nullable=True)
    current_core_business = Column(Text, nullable=True)
    detected_business_activities = Column(JSONB, nullable=True, default=list)
    detected_prohibited_activities = Column(JSONB, nullable=True, default=list)
    supporting_evidence = Column(JSONB, nullable=True, default=list)
    source_urls = Column(JSONB, nullable=True, default=list)
    source_publication_dates = Column(JSONB, nullable=True, default=list)
    ai_explanation = Column(Text, nullable=True)
    confidence_score = Column(Float, nullable=False, default=0.0)
    business_compliance_status = Column(String, nullable=True)
    last_analysed_timestamp = Column(DateTime(timezone=True), nullable=True)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
